qabel_provider/views.py

```diff
@@ -65,3 +57,39 @@ def get(self, request, format=None):
     def post(self, request, format=None):
         prefix = models.Prefix.objects.create(user=request.user)
         return Response(prefix.id, status=201)
+
+client = boto3.client('s3')
+transfer = boto3.s3.transfer.S3Transfer(client)
+s3 = boto3.resource('s3')
+
+
+@api_view(('GET', 'POST', 'DELETE'))
+@login_required()
+def file_resource(request, prefix, file_path, format=None):
+    if request.method == 'GET':
+        try:
+            with tempfile.NamedTemporaryFile('wb') as temp:
+                transfer.download_file(settings.BUCKET, '{}/{}'.format(prefix, file_path), temp.name)
+                temp.flush()
+                response = FileResponse(open(temp.name, 'rb'),
+                                        content_type='application/octet-stream')
+                return response
+        except ClientError:
+            raise Http404("File not found")
+    elif request.method == 'POST':
+        if prefix not in (str(p.id) for p in request.user.prefix_set.all()):
```

Comment:
Those parameters are enforced by the django request router defined in the urls.py